debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -187,10 +187,20 @@ private void allSlotsInROMHeaderDo() throws CorruptDataException {
 		srpCursor = romClass.innerClasses();
 		count = romClass.innerClassCount().intValue();
 		classWalkerCallback.addSection(clazz, srpCursor, count * SelfRelativePointer.SIZEOF, "innerClassesSRPs", true);
-		for (; count > 0; count--) {
+		for (; 0 != count; count--) {
 			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, srpCursor, "innerClassNameUTF8");
 			srpCursor = srpCursor.add(1);
 		}
+
+		/* walk enclosed inner classes SRPs block */
+		srpCursor = romClass.enclosedInnerClasses();
+		count = romClass.enclosedInnerClassCount().intValue();
+		classWalkerCallback.addSection(clazz, srpCursor, count * SelfRelativePointer.SIZEOF, "enclosedInnerClassesSRPs", true);
+		for (; 0 != count; count--) {
+			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, srpCursor, "enclosedInnerClassesNameUTF8");
+			srpCursor = srpCursor.add(1);
+		}
```

Comment:
This will need a `try/catch` block.

Reply:
Fixed as suggested.

Reply:
I don't see a `try/catch` block in the latest version (d8befbfdd811f3059748aa12fc949c2af7ec8451).

Reply:
Pushed the changes again.

Reply:
In d2cff15207cae6ed4466957ef8152f55dc269765 the `try` block is not around the calls that might throw `NoSuchFieldException`. Also, we don't want to catch all `Exception`s. It should be:
```
	try {
		/* walk enclosed inner classes SRPs block */
		srpCursor = romClass.enclosedInnerClasses();
		count = romClass.enclosedInnerClassCount().intValue();
		classWalkerCallback.addSection(clazz, srpCursor, count * SelfRelativePointer.SIZEOF, "enclosedInnerClassesSRPs", true);
		for (; 0 != count; count--) {
			classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, srpCursor, "enclosedInnerClassesNameUTF8");
			srpCursor = srpCursor.add(1);
		}
	} catch (NoSuchFieldException e) {
		// ignore: VM that generated the core dump doesn't have enclosedInnerClasses or enclosedInnerClassCount
	}
```
The methods `enclosedInnerClasses()` and `enclosedInnerClassCount()` declare that they throw `NoSuchFieldException`.

Reply:
Agreed and updated the code against the suggestion above given these fields don't always exist.